packages/graphile-build/src/SchemaBuilder.d.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  GraphQLSchema,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  GraphQLInputObjectType,
+  GraphQLEnumType,
+  GraphQLOutputType,
+  GraphQLFieldResolver,
+  GraphQLFieldConfigArgumentMap,
+  GraphQLFieldConfigMap,
+  GraphQLFieldConfig,
+  GraphQLEnumValueConfigMap,
+  GraphQLEnumValueConfig,
+  GraphQLInputFieldConfigMap,
+  GraphQLInputFieldConfig
+} from "graphql";
+import * as EventEmitter from "events";
+
+type mixed = {} | string | number | boolean | undefined | null;
+
+export interface Options {
+  [str: string]: mixed;
+}
+
+export type Plugin = (
+  builder: SchemaBuilder,
+  options: Options
+) => Promise<void> | void;
+
+type TriggerChangeType = () => void;
+export type WatchUnwatch = (triggerChange: TriggerChangeType) => void;
+
+export type SchemaListener = (newSchema: GraphQLSchema) => void;
+export type DataForType = {
+  [str: string]: Array<mixed>;
+};
+
+export type InitObject = {};
+
+export interface Build {
+  [str: string]: any;
+}
+export interface Inflection {
+  [str: string]: any;
+}
+export interface Scope<Type> {
+  [str: string]: any;
+}
+export interface Context<Type> {
+  scope: Scope<Type>;
+  [str: string]: any;
+}
+
+export type Hook<Type> = (
+  input: Type,
+  build: Build,
+  context: Context<Type>
+) => Type;
+
```

Comment:
Interesting; I know you can use interfaces for functions but I thought you typically only did that when they also have properties. Does using an interface for a vanilla function have advantages over a type alias? I used an alias because it seems simpler.